src/devices/Gpio/Drivers/RockPi4BPlusDriver.cs

```diff
@@ -13,26 +13,43 @@ namespace Iot.Device.Gpio.Drivers
     /// </remarks>
     public class RockPi4bPlusDriver : Rk3399Driver
     {
-        private static readonly int[] _pinNumberConverter = new int[]
-        {
-            -1, -1, MapPinNumber(2, 'A', 7), -1, MapPinNumber(2, 'B', 0), -1, MapPinNumber(2, 'B', 3),
-            MapPinNumber(4, 'C', 4), -1, MapPinNumber(4, 'C', 3), MapPinNumber(4, 'C', 2), MapPinNumber(4, 'A', 3),
-            MapPinNumber(4, 'C', 6), -1, MapPinNumber(4, 'C', 5), MapPinNumber(4, 'D', 2), -1, MapPinNumber(4, 'D', 4),
-            MapPinNumber(1, 'B', 0), -1, MapPinNumber(1, 'A', 7), MapPinNumber(4, 'D', 5), MapPinNumber(1, 'B', 1),
-            MapPinNumber(1, 'B', 2), -1, -1, MapPinNumber(2, 'A', 0), MapPinNumber(2, 'A', 1), MapPinNumber(2, 'B', 2),
-            -1, MapPinNumber(2, 'B', 1), MapPinNumber(3, 'C', 0), MapPinNumber(2, 'B', 4), -1, MapPinNumber(4, 'A', 5),
-            MapPinNumber(4, 'A', 4), MapPinNumber(4, 'D', 6), MapPinNumber(4, 'A', 6), -1, MapPinNumber(4, 'A', 7)
-        };
-
         /// <inheritdoc/>
         protected override int PinCount => 40;
 
         /// <inheritdoc/>
         protected override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber)
         {
-            int num = _pinNumberConverter[pinNumber];
-
-            return num != -1 ? num : throw new ArgumentException($"Board (header) pin {pinNumber} is not a GPIO pin on the {GetType().Name} device.", nameof(pinNumber));
+            return pinNumber switch
```

Comment:
Thanks for this improvement bu you still miss -1 which is a valid option. It means "no pin". So you should definitely add it.

Reply:
Really? That seems inconsistent with the rest of the code in this library. I can't find anywhere else where we treat -1 as a valid pin in a board numbering. Look at the same function in RaspberryPi3LinuxDriver for example.

Reply:
You are correct, I got catch by the previous implementation.